Assets/MixedRealityToolkit-SDK/Features/Input/FocusProvider.cs

```diff
@@ -313,12 +308,15 @@ public GameObject GetFocusedObject(IMixedRealityPointer pointingSource)
         /// <inheritdoc />
         public bool TryGetFocusDetails(IMixedRealityPointer pointer, out FocusDetails focusDetails)
         {
-            foreach (var pointerData in pointers)
+            if (pointer != null)
             {
-                if (pointerData.Pointer.PointerId == pointer.PointerId)
+                foreach (var pointerData in pointers)
```

Comment:
foreach -> for

Reply:
Pointers is a collection. For is not possible.

Reply:
![vaguegrippingantipodesgreenparakeet-size_restricted](https://user-images.githubusercontent.com/3580640/43996118-c22819a4-9d70-11e8-82be-784628fb4ec7.gif)

Reply:
But probably not advised :P

Reply:
https://stackoverflow.com/questions/9625270/what-is-the-fastest-safest-method-to-iterate-over-a-hashset

>A foreach loop has a small amount of addition overhead on an indexed collections (like an array). This is mostly because the foreach does a little more bounds checking than a for loop.
>
>HashSet does not have an indexer so you have to use the enumerator.
>
>In this case foreach is efficient as it only calls MoveNext() as it moves through the collection.